### PR TITLE
feat: IPEX apply, offer, agree endpoints

### PR DESCRIPTION
**Pull request by iFergal**
IPEX apply, offer, agree endpoints for non multi-sig AIDs. I can add multi-sig later if that's OK.

From https://github.com/WebOfTrust/keria/issues/193.

### CODE REVIEW

File: src/keria/app/ipexing.py [hunk 1]
```diff
@@ -266,3 +272,203 @@ def sendMultisigExn(agent, hab, ked, sigs, atc, rec):
...
+        # Now create the stream to send, need the signer seal
+        kever = hab.kever
+        seal = eventing.SealEvent(i=hab.pre, s="{:x}".format(kever.lastEst.s), d=kever.lastEst.d)
+
+        ims = eventing.messagize(serder=serder, sigers=sigers, seal=seal)
+
+        # make a copy and parse
+        agent.hby.psr.parseOne(ims=bytearray(ims))
+
+        agent.exchanges.append(dict(said=serder.said, pre=hab.pre, rec=rec, topic='credential'))
```
Comment (iFergal) [inline src/keria/app/ipexing.py:473]: I'm unsure if we are supposed to do something similar to admit (`Admitter`) where we now store the ACDC attachment that was offered, or if just the message is OK.

File: tests/app/test_ipexing.py [hunk 2]
```diff
@@ -919,3 +925,314 @@ def test_granter(helpers):
...
+        pre = aid['i']
+        assert pre == "EHgwVwQT15OJvilVvW57HE4w0-GPs_Stj2OFoAHZSysY"
+
+        salt2 = b'0123456789abcdeg'
+        op = helpers.createAid(client, "recp", salt2)
+        aid1 = op["response"]
+        pre1 = aid1['i']
+        assert pre1 == "EFnYGvF_ENKJ_4PGsWsvfd_R6m5cN-3KYsz_0mAuNpCm"
+
+        # This should be a metadata ACDC in reality
```
Comment (iFergal) [inline tests/app/test_ipexing.py:1034]: Question here... if a metadata ACDC exists for making commitments to the metadata of a private ACDC - does this mean for a public ACDC you offer the full public ACDC or still the metadata ACDC?

File: src/keria/app/ipexing.py (hunk 1 above)
Comment (pfeairheller) [reply src/keria/app/ipexing.py:473]: Just the message is OK here.

File: tests/app/test_ipexing.py (hunk 2 above)
Comment (pfeairheller) [reply tests/app/test_ipexing.py:1034]: I think it has to be the full public ACDC

**Review by pfeairheller — APPROVED** [review]

**Pull request merged**